fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/ProtocolSupport.java

```diff
@@ -0,0 +1,148 @@
+     */
+    public ProtocolSupport(
+            Collection<Protocol> protocols
+    ) {
+        this(protocols, Collections.emptySet());
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param protocols A collection of protocols this handler will support.
+     * @param blacklist  Protocols that this handler will neither support nor delegate.
+     */
+    public ProtocolSupport(
+            Collection<Protocol> protocols,
+            Collection<String> blacklist
+    ) {
+        protocolMap = protocols.stream().collect(Collectors.toMap(Protocol::getContractName, Function.identity()));
+        this.blacklist = blacklist;
+    }
+
+    /**
+     * Determine if this protocol handler or accepts this protocol.
+     *
+     * @param protocolName The name of the protocol to test for.
+     *
+     * @return TRUE if this metric directly or indirectly supports this signal, FALSE if it refuses, MAYBE if it
+     * doesn't assert authority.
+     */
+    public Accepts accepts(String protocolName) {
+        return protocolMap.containsKey(protocolName) ? Accepts.TRUE :
+                blacklist.contains(protocolName) ?
+                        Accepts.FALSE :
+                        Accepts.MAYBE;
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts a certain protocol.
+     *
+     * @param protocolName  The name of a protocol to not handle.
+     *
+     * @return A protocol handler with additional signals bound.
+     */
+    public ProtocolSupport withoutProtocol(String protocolName) {
+        return withoutProtocols(Collections.singleton(protocolName));
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts certain protocols.
+     *
+     * @param protocolNames  The names of protocols to not handle.
+     *
+     * @return A protocol handler with protocols not supported.
+     */
+    public ProtocolSupport withoutProtocols(Collection<String> protocolNames) {
+
+        List<Protocol> protocols =
+                protocolMap.values().stream()
+                        .filter(protocol -> !protocolNames.contains(protocol.getContractName()))
+                        .collect(Collectors.toList());
+        List<String> newBlackList = Stream.concat(protocolNames.stream(), blacklist.stream())
+                .collect(Collectors.toList());
+
+        return new ProtocolSupport(protocols, newBlackList);
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts protocols blacklisted in these other protocols.
+     *
+     * @param protocolSupport  The protocol supports whose blacklists should not be handled.
+     *
+     * @return A protocol handler with additional protocols not supported.
+     */
+    public ProtocolSupport combineBlacklists(Collection<ProtocolSupport> protocolSupport) {
+
+        List<String> protocols =
+                protocolSupport.stream()
+                        .flatMap(support -> support.blacklist.stream())
```

Comment:
Should we be using a (protected?) getter here, rather than accessing the field directly?

Reply:
meh.  direct access of members inside a class isn't the devil.